xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/transactions/AccountSet.java

```diff
@@ -65,25 +65,185 @@ default AccountSetTransactionFlags flags() {
+   * for unknown values, but binary serializing an {@link AccountSet} that was constructed by deserializing JSON would
+   * result in a different binary blob than what exists on ledger.</p>
+   *
+   * @return An {@link Optional} {@link UnsignedInteger}.
+   */
+  @JsonProperty("ClearFlag")
+  Optional<UnsignedInteger> clearFlagRawValue();
+
+  /**
+   * Normalization method to try to get {@link #clearFlag()}and {@link #clearFlagRawValue()} to match.
+   *
+   * <p>If neither field is present, there is nothing to do.</p>
+   * <p>If both fields are present, there is nothing to do, but we will check that {@link #clearFlag()}'s
+   * underlying value equals {@link #clearFlagRawValue()}.</p>
+   * <p>If {@link #clearFlag()} is present but {@link #clearFlagRawValue()} is empty, we set
+   * {@link #clearFlagRawValue()} to the underlying value of {@link #clearFlag()}.</p>
+   * <p>If {@link #clearFlag()} is empty and {@link #clearFlagRawValue()} is present, we will set
+   * {@link #clearFlag()} to the {@link AccountSetFlag} variant associated with {@link #clearFlagRawValue()}, or leave
+   * {@link #clearFlag()} empty if {@link #clearFlagRawValue()} does not map to an {@link AccountSetFlag}.</p>
+   *
+   * @return A normalized {@link AccountSet}.
+   */
+  @Value.Check
+  default AccountSet normalizeClearFlag() {
+    if (!clearFlag().isPresent() && !clearFlagRawValue().isPresent()) {
+      // If both are empty, nothing to do.
+      return this;
+    } else if (clearFlag().isPresent() && clearFlagRawValue().isPresent()) {
+      // Both will be present if:
+      //  1. A developer set them both manually (in the builder)
+      //  2. This normalize method has already been called.
+
+      // We should still check that the clearFlagRawValue matches the inner value of AccountSetFlag.
+      Preconditions.checkState(
+        clearFlag().get().getValue() == clearFlagRawValue().get().longValue(),
+        String.format("clearFlag and clearFlagRawValue should be equivalent, but clearFlag's underlying " +
+            "value was %s and clearFlagRawValue was %s",
+          clearFlag().get().getValue(),
+          clearFlagRawValue().get().longValue()
+        )
+      );
+      return this;
+    } else if (clearFlag().isPresent() && !clearFlagRawValue().isPresent()) {
+      // This can only happen if the developer only set clearFlag(). In this case, we need to set clearFlagRawValue to
+      // match clearFlag.
+      return AccountSet.builder().from(this)
+        .clearFlagRawValue(UnsignedInteger.valueOf(clearFlag().get().getValue()))
+        .build();
+    } else { // clearFlag not present and clearFlagRawValue is present
+      // This can happen if:
+      //   1. A developer sets clearFlag2 manually in the builder
+      //   2. JSON has ClearFlag and jackson sets clearFlag2.
+      if (clearFlagRawValue().get().longValue() <= 15) {
```

Comment:
I think zero could creep in here, no?

Reply:
It could, but `AccountSetFlag.NONE` exists and is mapped to zero so that should be fine.

Another way to do this is to call `AccountSetFlag.forValue` and catch the `IllegalArgumentException` as an indicator that the raw value isn't representable by `AccountSetFlag`